File: checkers/184/main.go
package main

import (
	"fmt"
	"testlib"
	"testlib/core"
)

type node struct {
	x1 int
	y1 int
	x2 int
	y2 int
	l  bool
}

func B2i(b bool) int {
	if b {
		return 1
	}
	return 0
}

func main() {
	defer func() {
		_ = testlib.GetStream(testlib.TaskOutput).Close()
		_ = testlib.GetStream(testlib.InputFromUser).Close()
	}()
	inputStream := testlib.GetReader(testlib.TaskInput)
	outputStream := testlib.GetReader(testlib.TaskOutput)
	inputFromUserStream := testlib.GetReader(testlib.InputFromUser)

	n1, n2 := core.ReadInt(inputFromUserStream), core.ReadInt(outputStream)

	if n1 > n2 {
		testlib.WriteAnswer(fmt.Sprintf("WA"), testlib.WA)
		return
	}
	if n1 == -1 && n2 == -1 {
		testlib.WriteAnswer("OK", testlib.OK)
		return
	}

	// n, m := 8, 8
	x1, y1, x2, y2 := core.ReadInt(inputStream), core.ReadInt(inputStream), core.ReadInt(inputStream), core.ReadInt(inputStream)
	now := node{
		x1: x1,
		y1: y1,
		x2: x2,
		l:  true,
	}
	f := false
	ff := false
	l3, x3, y3 := 0, 0, 0
	dx, dy := make([]int, 0), make([]int, 0)
	for i := 0; i < n1; i++ {
		l3, x3, y3 = core.ReadInt(inputFromUserStream), core.ReadInt(inputFromUserStream), core.ReadInt(inputFromUserStream)
		l3--
		if l3 == 1 {
			if B2i(now.l) == l3 {
				testlib.WriteAnswer(fmt.Sprintf("WA1"), testlib.WA)
				return
			}
			dx = append(dx, 2, 2, -2, -2, 1, 1, -1, -1)
			dy = append(dy, -1, 1, -1, 1, -2, 2, -2, 2)
			ff = false
			for i := 0; i < 8; i++ {
				if now.x2-x3 == dx[i] && now.y2-y3 == dy[i] {
					ff = true
				}
			}
			if ff == false {
				testlib.WriteAnswer(fmt.Sprintf("WA2"), testlib.WA)
				return
			}
		} else {
			if B2i(now.l) == l3 {
				testlib.WriteAnswer(fmt.Sprintf("WA3"), testlib.WA)
				return
			}
			dx = append(dx, 2, 2, -2, -2, 1, 1, -1, -1)
			dy = append(dy, -1, 1, -1, 1, -2, 2, -2, 2)
			ff = false
			for i := 0; i < 8; i++ {
				if now.x1-x3 == dx[i] && now.y1-y3 == dy[i] {
					ff = true
				}
			}
			if B2i(ff) == 0 {
				testlib.WriteAnswer(fmt.Sprintf("WA4"), testlib.WA)
				return
			}
		}
		if l3 == 1 {
			now.x2 = x3
			now.y2 = y3
		} else {
			now.x1 = x3
			now.y1 = y3
		}
		now.l = !(now.l)
		// cleaning slice:
		dx = dx[:0]
		dy = dy[:0]
	}
	if B2i(f) == 1 {
		testlib.WriteAnswer(fmt.Sprintf("WA5"), testlib.WA)
		return
	}
	if now.x1 != x2 || now.y1 != y2 || now.x2 != x1 || now.y2 != y1 {
		testlib.WriteAnswer(fmt.Sprintf("WA6"), testlib.WA)
		return
	}
	testlib.WriteAnswer("OK", testlib.OK)
}

File: checkers/184/main.cpp
#include <checkers/testlib.h>
#include <iostream>
#include <vector>
#include <set>
#include <algorithm>
#include <map>
#include <string>
#include <sstream>

using namespace NTestlib;

struct node
{
    int x1, y1, x2, y2;
    bool l;
};

int main(int argc, char *argv[])
{
    InitChecker(argc, argv);
    int n1 = Out.ReadInt();
    int n2 = Ans.ReadInt();
    if (n1 > n2)
        QuitWith(WA, "WA");
    if (n1 == -1 && n2 == -1)
        QuitWith(AC, "OK");
    int n = 8;
    int m = 8;
    int x1 = File.ReadInt();
    int y1 = File.ReadInt();
    int x2 = File.ReadInt();
    int y2 = File.ReadInt();
    node now = {x1, y1, x2, y2, 1};
    bool f = 0;
    for (int i = 0; i < n1; i++)
    {
        int l3 = Out.ReadInt();
        int x3 = Out.ReadInt();
        int y3 = Out.ReadInt();
        l3--;
        if (l3 == 1)
        {
            if (now.l == l3)
                QuitWith(WA, "WA1");
            int dx[8] = {2, 2, -2, -2, 1, 1, -1, -1};
            int dy[8] = {-1, 1, -1, 1, -2, 2, -2, 2};
            bool ff = 0;
            for (int i = 0; i < 8; i++)
                if (now.x2 - x3 == dx[i] && now.y2 - y3 == dy[i])
                    ff = 1;
            if (ff = 0)
                QuitWith(WA, "WA2");
        }
        else
        {
            if (now.l == l3)
                QuitWith(WA, "WA3");
            int dx[8] = {2, 2, -2, -2, 1, 1, -1, -1};
            int dy[8] = {-1, 1, -1, 1, -2, 2, -2, 2};
            bool ff = 0;
            for (int i = 0; i < 8; i++)
                if (now.x1 - x3 == dx[i] && now.y1 - y3 == dy[i])
                    ff = 1;
            if (ff = 0)
                QuitWith(WA, "WA4");
        }
        if (l3 == 1)
        {
            now.x2 = x3;
            now.y2 = y3;
        }
        else
        {
            now.x1 = x3;
            now.y1 = y3;
        }
        now.l = !now.l;
    }
    if (f == 1)
        QuitWith(WA, "WA5");
    if (now.x1 != x2 || now.y1 != y2 || now.x2 != x1 || now.y2 != y1)
        QuitWith(WA, "WA6");
    QuitWith(AC, "OK");
}